src/blocks/remote-data-container/index.ts

```diff
@@ -52,11 +56,109 @@ addFilter(
  */
 addFilter( 'blocks.registerBlockType', 'remote-data-blocks/addUsesContext', addUsesContext, 10 );
 
+interface State {}
+
+type Actions = {
+	GET_DATA: () => void;
+};
+
+interface Selectors {}
+
+const queryDataStateKey = (queryKey: string, blockName: string, queryInputs: Record<string, string>) => `${blockName}:${queryKey}:${JSON.stringify(queryInputs ?? {})}`;
+
+const remoteDataBlocksStoreConfig: ReduxStoreConfig< State, Actions, Selectors > = {
+	reducer: ( state = {}, action ) => {
+		switch ( action.type ) {
+			case 'RECEIVE_REMOTE_DATA':
+				// console.log('store: RECEIVED REMOTE DATA')
+				// console.log({action});
+				// console.log(`key: ${queryDataStateKey(action.queryKey, action.blockName, action.queryInput)}`);
+				return { ...state, [ queryDataStateKey(action.queryKey, action.blockName, action.queryInputs) ]: action.data };
+			case 'RECEIVE_REMOTE_DATA_ERROR':
+				// console.log('store: RECEIVED REMOTE DATA ERROR')
+				return { ...state, [ queryDataStateKey(action.queryKey, action.blockName, action.queryInputs) ]: action.error };
+		}
+		return state;
+	},
+	selectors: {
+		getRemoteData: ( state, queryKey, blockName, queryInputs = {} ) => {
+			// console.log( 'store: CALLED SELECTOR' );
+			console.log( { state, queryKey, blockName, queryInputs } );
+			return state[ queryDataStateKey(queryKey, blockName, queryInputs) ];
+		},
+	},
+	resolvers: {
+		getRemoteData:
+			( queryKey: string, blockName: string, queryInputs: Record< string, string > ) =>
+			async ( { dispatch } ) => {
+				console.log( 'store: CALLED RESOLVER', { queryKey, blockName, queryInputs } );
+				try {
+						console.log({FETCH:{blockName, queryKey, queryInputs }});
+					const data = await fetchRemoteData( {
+						block_name: blockName,
+						query_key: queryKey,
+						query_inputs: [	queryInputs	],
+					} );
+					// console.log('store: DISPATCHING RECEIVE_REMOTE_DATA')
+					console.log( { retrievedData: data } );
+					dispatch( { type: 'RECEIVE_REMOTE_DATA', blockName, queryKey, data, queryInputs } );
+				} catch ( err: unknown ) {
+					console.error(err);
+					dispatch( { type: 'RECEIVE_REMOTE_DATA_ERROR', blockName, queryKey, error: err } );
+				}
+			},
+	},
+};
+
+const remoteDataBlocksStore = createReduxStore(
+	'remote-data-blocks-store',
+	remoteDataBlocksStoreConfig
+);
+
+register( remoteDataBlocksStore );
+
 registerBlockBindingsSource( {
 	name: 'remote-data/binding',
 	label: 'Remote Data Binding',
 	usesContext: [ 'remote-data-blocks/remoteData' ],
-	getValues() {
-		return {};
+	getValues( { context, clientId, bindings, select, ...other } ) {
```

Comment:
Should we be using `setValues` too? Or should we be dispatching actions from `useRemoteData()`?